utils/frame/rpc/state-trie-migration-rpc/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+		let (key, value) = key_value.map_err(|e| format!("TrieDB node iterator error: {}", e))?;
+		if key[..].starts_with(sp_core::storage::well_known_keys::DEFAULT_CHILD_STORAGE_KEY_PREFIX)
+		{
+			let prefixed_key = PrefixedStorageKey::new(key);
+			let (_type, unprefixed) = ChildType::from_prefixed_key(&prefixed_key).unwrap();
+			child_roots.push((ChildInfo::new_default(unprefixed), value));
+		}
+	}
+	for (child_info, root) in child_roots {
+		let mut child_root = H::Out::default();
+		let storage = KeySpacedDB::new(essence, child_info.keyspace());
+
+		child_root.as_mut()[..].copy_from_slice(&root[..]);
+		let trie = sp_trie::trie_types::TrieDB::new(&storage, &child_root)
+			.map_err(|e| format!("New child TrieDB error: {}", e))?;
+		let iter_node = TrieDBNodeIterator::new(&trie)
+			.map_err(|e| format!("TrieDB node iterator error: {}", e))?;
+		for node in iter_node {
+			let node = node.map_err(|e| format!("Child TrieDB node iterator error: {}", e))?;
+			match node.2.node_plan() {
+				NodePlan::Leaf { value, .. } |
+				NodePlan::NibbledBranch { value: Some(value), .. } =>
+					if let ValuePlan::Inline(range) = value {
+						if (range.end - range.start) as u32 >= threshold {
+							nb_to_migrate_child += 1;
+						}
+					},
+				_ => (),
+			}
+		}
+	}
+
+	Ok((nb_to_migrate, nb_to_migrate_child))
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+pub struct MigrationStatusResult {
+	top_remaining_to_migrate: u64,
+	child_remaining_to_migrate: u64,
+}
+
+/// Migration RPC methods.
+#[rpc]
+pub trait StateMigrationApi<BlockHash> {
+	/// Check current migration state.
+	///
+	/// This call is performed locally without submitting any transactions. Thus executing this
+	/// won't change any state. Nonetheless it is a VERY costy call that should be
+	/// only exposed to trusted peers.
+	#[rpc(name = "state_trieMigrationStatus")]
+	fn call(&self, at: Option<BlockHash>) -> Result<MigrationStatusResult>;
+}
+
+/// An implementation of state migration specific RPC methods.
+pub struct MigrationRpc<C, B, BA> {
+	client: Arc<C>,
+	backend: Arc<BA>,
+	deny_unsafe: DenyUnsafe,
+	_marker: std::marker::PhantomData<(B, BA)>,
+}
+
+impl<C, B, BA> MigrationRpc<C, B, BA> {
+	/// Create new state migration rpc for the given reference to the client.
+	pub fn new(client: Arc<C>, backend: Arc<BA>, deny_unsafe: DenyUnsafe) -> Self {
+		MigrationRpc { client, backend, deny_unsafe, _marker: Default::default() }
+	}
+}
+
+impl<C, B, BA> StateMigrationApi<<B as BlockT>::Hash> for MigrationRpc<C, B, BA>
+where
+	B: BlockT,
+	C: Send + Sync + 'static + sc_client_api::HeaderBackend<B>,
+	BA: 'static + sc_client_api::backend::Backend<B>,
+{
+	fn call(&self, at: Option<<B as BlockT>::Hash>) -> Result<MigrationStatusResult> {
+		if let Err(err) = self.deny_unsafe.check_if_safe() {
+			return Err(err.into())
+		}
+
+		let block_id = BlockId::hash(at.unwrap_or_else(|| self.client.info().best_hash));
```

Comment:
```suggestion
		let block_id = BlockId::hash(at.unwrap_or(self.client.info().best_hash));
```

Reply:
I like my unwrap_or_else, not that I don't trust the compiler, but I find it more explicit :) I will keep it if it's not an issue.